apm-agent-core/src/main/java/co/elastic/apm/agent/bci/ElasticApmAgent.java

```diff
@@ -505,7 +505,8 @@ public static void validateAdvice(ElasticApmInstrumentation instrumentation) {
         for (MethodDescription.InDefinedShape exitAdvice : typeDescription.getDeclaredMethods().filter(isStatic().and(isAnnotatedWith(Advice.OnMethodExit.class)))) {
             validateAdviceReturnAndParameterTypes(exitAdvice, adviceClassName);
             validateLegacyAssignToIsNotUsed(exitAdvice);
-            if (exitAdvice.getReturnType().asRawType().getTypeName().startsWith("co.elastic.apm")) {
+            String returnType = exitAdvice.getReturnType().asRawType().getTypeName();
+            if (returnType.startsWith("co.elastic.apm") && !returnType.startsWith("co.elastic.apm.api")) {
```

Comment:
Let's continue disallowing usage of API types within advices

Reply:
Done